audformat/core/utils.py

```diff
@@ -640,6 +641,62 @@ def same_dtype(d1, d2) -> bool:
     return d1.name == d2.name
 
 
+def set_file_extension(
+        index: pd.Index,
+        extension: str,
+        pattern: str = r'\.[a-zA-Z0-9]+$',
+) -> pd.Index:
+    r"""Change the file extension of index entries.
+
+    It replaces all existing file extensions
+    in the index file path
+    by the new provided one.
+
+    Args:
+        index: index with file path
+            conform to :ref:`table specifications <data-tables:Tables>`
+        extension: new file extension without ``'.'``.
+            If set to ``''``,
+            the current file extension is removed
+        pattern: regexp pattern to match current extensions.
+            In contrast to ``extension``,
+            you have to include ``'.'``
+
+    Returns:
+        updated index
+
+    Example:
+        >>> idx = filewise_index(['f1.wav', 'f2.flac'])
+        >>> set_file_extension(idx, 'mp3')
+        Index(['f1.mp3', 'f2.mp3'], dtype='object', name='file')
+        >>> idx = filewise_index(['f1.wav.gz', 'f2.wav.gz'])
+        >>> set_file_extension(idx, '')
+        Index(['f1.wav', 'f2.wav'], dtype='object', name='file')
+        >>> set_file_extension(idx, 'flac', pattern='.wav.gz')
```

Comment:
For best practise maybe change to ` pattern='\.wav\.gz$'?

Reply:
Good that you reminded me of that one. I changed to `pattern=r'\.wav\.gz$'`.

Reply:
In principle, the function can now be used to replace any parts of the file path. But I guess it still makes sense to stick keep the name `audformat.utils.set_file_extension()` as this is the most likely use case.

Reply:
I would now also be fine with using `replace_file_extension()`, but I don't have an opinion which is better.

Reply:
I think I would prefer `replace_file_extension()` now.

Reply:
Renamed it